### DISCUSSION TITLE
Inability to select compiler kit when using my preset

### DISCUSSION THREAD

**Discussion by TommasoTricker** [microsoft/vscode-cmake-tools#4200] [Q&A] [answered]
How do i select a compiler kit when having my preset active? Normally I get prompted upon pressing the build button for the first time but now my build fails with "No CMAKE_CXX_COMPILER could be found.", as I haven't been able to select it.
```
{
    "version": 2,
    "configurePresets": [
        {
            "name": "vcpkg",
            "generator": "Ninja",
            "binaryDir": "${sourceDir}/build",
            "cacheVariables": {
                "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
            }
        }
    ]
}
```

**Comment by gcampbell-msft** [comment, marked as answer]
@TommasoTricker When using presets, you should use the preset itself to specify your compiler, most often through the CMAKE_C_COMPILER and CMAKE_CXX_COMPILER cache variables in your preset. Thanks!
